conversion/conversion.go

```diff
@@ -201,27 +255,103 @@ func snapshotMigrationHandler(sourceProfile profiles.SourceProfile, config write
+			return dataFromDatabaseForDataflowMigration(targetProfile, ctx, sourceProfile, conv)
+		} else if sourceProfile.Config.ConfigType == constants.DMS_MIGRATION {
+			return dataFromDatabaseForDMSMigration()
+		} else {
+			return nil, fmt.Errorf("configType should be one of 'bulk', 'dataflow' or 'dms'")
+		}
+	default:
+		infoSchema, err := GetInfoSchema(sourceProfile, targetProfile)
 		if err != nil {
 			return nil, err
 		}
-		bw, err := snapshotMigrationHandler(sourceProfile, config, conv, client, infoSchema)
+		var streamInfo map[string]interface{}
+		if sourceProfile.Conn.Streaming {
+			streamInfo, err = infoSchema.StartChangeDataCapture(ctx, conv)
+			if err != nil {
+				return nil, err
+			}
+			bw, err := snapshotMigrationHandler(sourceProfile, config, conv, client, infoSchema)
+			if err != nil {
+				return nil, err
+			}
+			err = infoSchema.StartStreamingMigration(ctx, client, conv, streamInfo)
+			if err != nil {
+				return nil, err
+			}
+			return bw, nil
+		}
+		return performSnapshotMigration(config, conv, client, infoSchema), nil
+	}
+}
+
+// TODO: Define the data processing logic for DMS migrations here.
+func dataFromDatabaseForDMSMigration() (*writer.BatchWriter, error) {
+	return nil, fmt.Errorf("dms configType is not implemented yet, please use one of 'bulk' or 'dataflow'")
+}
+
+//1. Create batch for each physical shard
+//2. Create streaming cfg from the config source type.
+//3. Verify the CFG and update it with HB defaults
+//4. Launch the stream for the physical shard
+//5. Perform streaming migration via dataflow
+func dataFromDatabaseForDataflowMigration(targetProfile profiles.TargetProfile, ctx context.Context, sourceProfile profiles.SourceProfile, conv *internal.Conv) (*writer.BatchWriter, error) {
+	asyncProcessShards := func(p *profiles.DataShard, mutex *sync.Mutex) common.TaskResult[*profiles.DataShard] {
+
+		streamingCfg := streaming.CreateStreamingConfig(*p)
+
+		err := streaming.VerifyAndUpdateCfg(&streamingCfg, targetProfile.Conn.Sp.Dbname)
+		if err != nil {
+			err = fmt.Errorf("failed to process shard: %s, there seems to be an error in the sharding configuration, error: %v", p.DataShardId, err)
+			return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
+		}
+		fmt.Printf("Initiating migration for shard: %v\n", p.DataShardId)
+
+		err = streaming.LaunchStream(ctx, sourceProfile.Driver, p.LogicalShards, targetProfile.Conn.Sp.Project, streamingCfg.DatastreamCfg)
 		if err != nil {
-			return nil, err
+			return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
 		}
-		err = infoSchema.StartStreamingMigration(ctx, client, conv, streamInfo)
+
+		err = streaming.StartDataflow(ctx, targetProfile, streamingCfg, conv)
+		return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
+	}
+	_, err := common.RunParallelTasks(sourceProfile.Config.ShardConfigurationDataflow.DataShards, 5, asyncProcessShards, true)
```

Comment:
Better to define 5 as constant above so that its easier to find and modify the value of parallelisation

Reply:
Done